### ISSUE TITLE
Rule: Warn for "new" capitalization

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#4]
This rule should warn when `new` is used prior to a function that doesn't have a capital letter as the first letter, such as:

``` js
var obj = new object();
```

Rule name should be `new-cap`.
